TopMovies/Core/State/MoviesState.swift

```diff
@@ -18,6 +18,14 @@ extension MoviesState {
     switch action {
     case let MoviesDownloadingAction.completed(_, movies):
       return MoviesState(relational: movies.hashMap(into: state.relational, id: \.id))
+    case let moviesCategories as CompletedMovieCategoriesAction:
+      return MoviesState(relational: state.relational
+                          .merging(moviesCategories.moviesRelational)
+                            { $1 })
+    case let movies as CompletedMoviesListAction:
+      return MoviesState(relational: state.relational
+                          .merging(movies.list.hashMap(into: state.relational, id: \.id))
+                            { $1 })
```

Comment:
Remove of `merging` function since you've already merged movies array into relational:
```swift
return MoviesState(
  relational: movies.list.hashMap(into: state.relational, id: \.id)
)
```